contracts/adapters/swap/oraidex/src/helper.rs

```diff
@@ -52,3 +57,73 @@ pub fn convert_pool_id_to_v3_pool_key(pool_id: &str) -> Result<PoolKey, Contract
         fee_tier: FeeTier { fee, tick_spacing },
     })
 }
+
+pub fn parse_to_swap_msg(
+    deps: &Deps,
+    operation: SwapOperation,
+) -> Result<(Addr, Binary), ContractError> {
+    // case 1: convert
+    if operation.pool.contains("convert") {
+        let parts: Vec<&str> = operation.pool.split('-').collect();
+        if parts.len() != 2 {
+            return Err(ContractError::Std(StdError::generic_err(
+                "Invalid convert type pool_id, require exactly 2 fields",
+            )));
+        }
+        let converter = deps.api.addr_validate(parts[1])?;
+
+        match parts[0] {
+            "convert_reverse" => match deps.api.addr_validate(&operation.denom_in) {
+                Ok(_addr) => {
+                    return Ok((
+                        converter,
+                        to_json_binary(&ConverterCw20HookMsg::ConvertReverse {
+                            from: denom_to_asset_info(deps.api, &operation.denom_out),
+                        })?,
+                    ));
+                }
+                Err(_) => {
+                    return Ok((
+                        converter,
+                        to_json_binary(&ConverterExecuteMsg::ConvertReverse {
+                            from_asset: denom_to_asset_info(deps.api, &operation.denom_out),
+                        })?,
+                    ));
+                }
+            },
+            "convert" => {
+                return Ok((converter, to_json_binary(&ConverterExecuteMsg::Convert {})?));
+            }
+            _ => {
+                return Err(ContractError::Std(StdError::generic_err(
+                    "Invalid convert type pool_id",
+                )));
+            }
+        }
+    }
+
+    let mut hop_swap_requests: Vec<OraidexSwapOperation> = vec![];
+    let oraidex_router_contract_address = ORAIDEX_ROUTER_ADDRESS.load(deps.storage)?;
+
+    // case 2: Swap v3
+    if operation.pool.contains("-") {
+        let pool_key = convert_pool_id_to_v3_pool_key(&operation.pool)?;
+        let x_to_y = pool_key.token_x == operation.denom_in;
+        hop_swap_requests.push(OraidexSwapOperation::SwapV3 { pool_key, x_to_y });
+    } else {
```

Comment:
Should we specifically check if the operation is swap v2? A fallback to v2 can be risky. We can fallback to an error if the operation is neither v3 or v2

Reply:
we default swap through v2 if poolID is not of v3 format. Pool V2 only needs denom_in & denom_out to determine pool, and if this pool doesn't exist, it will throw pool not found in factory